doc/api/external_formats.md

```diff
@@ -0,0 +1,78 @@
+# Modules: External formats
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+## Introduction
+
+Node natively understands a handful of formats (see the table in [Modules: Load hook][Load
+customization hook]). Non-native / external formats require a translator (under the hood, this is a
+custom loader). When attempting to run a module with an external format (ex `node main.ts`), node
+will try to detect what it is. At current, when node is able to identify the format, it prints a
+message to stdout with instructions (that lead here).
+
+## Setting up a module translator
+
+There are a few, currently manual, steps you need to complete. We intend to automate these in
+future.
+
+1. You will need to find and install a module translator via your preferred package manager. Some
+   package managers provide a CLI utility as well as a website.
+1. Once installed, in order to get node to automatically use it, you will need to create a `.env`
+   file containing a [`--import`][CLI import] for your chosen module translator's package (ex `NODE_OPTIONS="--import=example-translator/register"`). See [CLI:
+   env file][CLI env file].
+1. When you subsequently run node, include the env file flag (ex `node --env-file=.env main.ts`).
+
+## Publishing a module translator
+
+A few specific things are necessary when publishing a module translator.
+
+### Package registration
+
+Your package registration needs the following tags (aka keywords): `module translator`, `node`, and
```

Comment:
> `nodejs-external-format-support`

😬 🙅‍♂️

Reply:
> And tell them where to put these; in package.json keywords?

I don't know how these get assigned to the npm registration—be they extracted from package.json or explicitly configured with npm. I think that's outside the scope of this doc (and should be handed by npm's docs).

Reply:
`npm` isn’t going to provide any help with this. We need to work within what they already support, which is the `keywords` field in `package.json`.

Reply:
Ah, okay. I wasn't expecting any help with it since they already support the keyword search (that's all we need). If those come from package.json, sure, all the simpler 🙂